src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -148,29 +149,33 @@ protected function getEffectivePlugins(array $resources): array
      */
     protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
     {
-        foreach ($effectivePlugins as $effectivePlugin) {
+        foreach ($effectivePlugins as $effectiveEventPlugins) {
+            $effectivePlugin = $this->resolveDuplicatePlugins($effectiveEventPlugins);
```

Comment:
Please, let me know. If you have case, when $effectiveEventPlugins have to plugins:
1) ie `instance of EventResourceRepositoryPluginInterface`
2) ie `instance of EventResourceQueryContainerPluginInterface`

Why do you leave only the first one?
It looks like `resolveDuplicatePlugins` should have the name `resolvePriorityPlugins`.

But maybe rewrite `resolveDuplicatePlugins` and return array with plugins, if there will for example `EventResourceQueryContainerPluginInterface` and `EventResourceRepositoryPluginInterface`, what do you think?

Reply:
It's not exactly `Priority` based, maybe Preferability. but that's what `resolveDuplicatePlugins` means, if there's duplicates, don't just pick the first one as before. pick X if exists otherwise pick the first one. If both duplicates are of type X, it will also pick the first one.

Reply:
why we don't need all the plugins?

Reply:
As for me, we could use not only one plugin, what do you think?

Reply:
Because that's what this was and is about, eliminating duplicate plugins for the same event and resource.

if an event for a resource had two plugins (which we have already) don't push duplicate messages to the queue.

Reply:
If this does not change the behaviour, then OK